Sources/Host/POSIX/Platform.cpp

```diff
@@ -124,6 +127,23 @@ bool Platform::SetWorkingDirectory(std::string const &directory) {
 
 ds2::ProcessId Platform::GetCurrentProcessId() { return ::getpid(); }
 
+// Termiantes the process and waits for it to exit before returning.
+bool Platform::TerminateProcess(ProcessId pid) {
+  if (::kill(pid, SIGKILL) < 0)
+    return false;
+
+  int status;
+  pid_t ret;
+  do {
+    ret = ::waitpid(pid, &status, 0);
```

Comment:
are there any concerns with infinite hangs if waitpid will wait indefinetely here, for example if the child process is stuck? Not sure if it's ever a concern, just curious